sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -39,29 +40,346 @@ class ConfigurationSetting(Model):
+            label=self.label,
+            value=self._value,
+            content_type=self.content_type,
+            last_modified=self.last_modified,
+            tags=self.tags,
+            locked=self.read_only,
+            etag=self.etag,
+        )
+
+
+class SecretReferenceConfigurationSetting(ConfigurationSetting):
+    """A configuration value that references a KeyVault Secret
+    Variables are only populated by the server, and will be ignored when
+    sending a request.
+    :ivar etag: Entity tag (etag) of the object
+    :vartype etag: str
+    :ivar key:
+    :vartype key: str
+    :ivar secret_uri:
+    :vartype secret_uri: str
+    :param label:
+    :type label: str
+    :param content_type:
+    :type content_type: str
+    :ivar value: The value of the configuration setting
+    :vartype value: str
+    :ivar last_modified:
+    :vartype last_modified: datetime
+    :ivar read_only:
+    :vartype read_only: bool
+    :param tags:
+    :type tags: dict[str, str]
+    """
+
+    _attribute_map = {
+        "etag": {"key": "etag", "type": "str"},
+        "key": {"key": "key", "type": "str"},
+        "label": {"key": "label", "type": "str"},
+        "content_type": {"key": "content_type", "type": "str"},
+        "value": {"key": "value", "type": "str"},
+        "last_modified": {"key": "last_modified", "type": "iso-8601"},
+        "read_only": {"key": "read_only", "type": "bool"},
+        "tags": {"key": "tags", "type": "{str}"},
+    }
+    _secret_reference_content_type = (
+        "application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8"
+    )
+    kind = "SecretReference"
+
+    def __init__(self, key, secret_uri, label=None, **kwargs):
+        # type: (str, str, str) -> None
+        self._secret_uri = secret_uri
+        super(SecretReferenceConfigurationSetting, self).__init__(**kwargs)
+        self.key = key
+        self.label = label
+        self.content_type = kwargs.get(
+            "content_type", self._secret_reference_content_type
+        )
+        self.etag = kwargs.get("etag", None)
+        self.last_modified = kwargs.get("last_modified", None)
+        self.read_only = kwargs.get("read_only", None)
+        self.tags = kwargs.get("tags", {})
+        self._value = {"secret_uri": self._secret_uri}
+
+    @property
+    def secret_uri(self):
+        # type: () -> str
+        self._validate()
+        return self._value['secret_uri']
+
+    @secret_uri.setter
+    def secret_uri(self, value):
+        if self._value is None or isinstance(self._value, dict):
+            if self._value is None:
+                self._value = {}
+            self._value["secret_uri"] = value
+        else:
+            raise ValueError("Expect 'value' to be a dictionary.")
+
+    def _validate(self):
+        # type: () -> None
+        if not (self._value is None or isinstance(self._value, dict)):
+            raise ValueError("Expect 'value' to be a dictionary or None.")
+
+    @property
```

Comment:
Given

```py
@property
    def value(self):
        # type: () -> Dict[str, Any]
        return self._value

    @value.setter
    def value(self, value):
        # type: (Dict[str, Any]) -> None
        self._value = value
```
Maybe we don't need _value. :)

Reply:
In fact, after I re-think about it. ConfigSetting.value is a string. why here it is a dict?

Should we return a string in getter?

Reply:
It is stored as json in the service I think it makes the most sense to treat it as a dictionary to make it the most valuable to the user.